### ISSUE TITLE
Garbage collection of symbols

### ISSUE THREAD

**Issue by mnieper** [ashinn/chibi-scheme#488]
The following code causes Chibi to allocate more and more memory:

```scheme
(do ((i 0 (+ i 1)))
     (#f)
   (symbol->string (string->symbol (number->string i)))
   (when (zero? (modulo i 10000))
     (display (gc))
     (newline)))
```

Aren't the generated interned symbols garbage-collected?

**Comment by ashinn** [comment]
Yes, the symbol table implementation predates weak references.  We can probably fix this now easily.

**Comment by mnieper** [comment]
A slight complication may arise when symbols are used as weak keys (for example, in ephemerons). As symbols can always be reforged using `string->symbol` a symbol weak key may never be invalidated even if the interned symbol is being garbage collected.
